Modules/_operator.c

```diff
@@ -1580,30 +1654,28 @@ methodcaller_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         return NULL;
     }
 
-    name = PyTuple_GET_ITEM(args, 0);
     Py_INCREF(name);
     PyUnicode_InternInPlace(&name);
     mc->name = name;
 
+    mc->xargs = Py_XNewRef(args); // allows us to use borrowed references
     mc->kwds = Py_XNewRef(kwds);
+    mc->vectorcall_args = 0;
 
-    mc->args = PyTuple_GetSlice(args, 1, PyTuple_GET_SIZE(args));
-    if (mc->args == NULL) {
-        Py_DECREF(mc);
-        return NULL;
-    }
+
+    mc->vectorcall = (vectorcallfunc)methodcaller_vectorcall;
 
     PyObject_GC_Track(mc);
     return (PyObject *)mc;
 }
 
-static int
+static void
 methodcaller_clear(methodcallerobject *mc)
 {
     Py_CLEAR(mc->name);
-    Py_CLEAR(mc->args);
+    Py_CLEAR(mc->xargs);
     Py_CLEAR(mc->kwds);
-    return 0;
+    _methodcaller_clear_vectorcall(mc);
```

Comment:
Why did you separate the function? 
Please, just inline the function in here if there is no specific reason.

Reply:
To make it more symmetric with the `_methodcaller_initialize_vectorcall`. I will inline here. Let me know if the `_methodcaller_initialize_vectorcall` should also be inlined.